src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AppleCCCryptor.cs

```diff
@@ -0,0 +1,248 @@
+        }
+
+        private unsafe int CipherUpdate(byte[] input, int inputOffset, int count, byte[] output, int outputOffset)
+        {
+            int ret;
+            int errorCode;
+            int bytesWritten;
+
+            if (count == 0)
+            {
+                return 0;
+            }
+
+            fixed (byte* inputStart = input)
+            fixed (byte* outputStart = output)
+            {
+                byte* inputCurrent = inputStart + inputOffset;
+                byte* outputCurrent = outputStart + outputOffset;
+
+                ret = Interop.AppleCrypto.CryptorUpdate(
+                    _cryptor,
+                    inputCurrent,
+                    count,
+                    outputCurrent,
+                    output.Length - outputOffset,
+                    out bytesWritten,
+                    out errorCode);
+            }
+
+            Exception exception = ProcessInteropError(ret, errorCode);
+
+            if (exception != null)
+            {
+                throw exception;
+            }
+
+            return bytesWritten;
+        }
+
+        private unsafe void OpenCryptor(
+            Interop.AppleCrypto.PAL_SymmetricAlgorithm algorithm,
+            CipherMode cipherMode,
+            byte[] key)
+        {
+            int ret;
+            int errorCode;
+
+            byte[] iv = IV;
+
+            fixed (byte* pbKey = key)
+            fixed (byte* pbIv = iv)
+            {
+                ret = Interop.AppleCrypto.CryptorCreate(
+                    _encrypting
+                        ? Interop.AppleCrypto.PAL_SymmetricOperation.Encrypt
+                        : Interop.AppleCrypto.PAL_SymmetricOperation.Decrypt,
+                    algorithm,
+                    GetPalChainMode(cipherMode),
+                    Interop.AppleCrypto.PAL_PaddingMode.None,
+                    pbKey,
+                    key.Length,
+                    pbIv,
+                    Interop.AppleCrypto.PAL_SymmetricOptions.None,
+                    out _cryptor,
+                    out errorCode);
+            }
+
+            Exception exception = ProcessInteropError(ret, errorCode);
+
+            if (exception != null)
+            {
+                throw exception;
+            }
+        }
+
+        private Interop.AppleCrypto.PAL_ChainingMode GetPalChainMode(CipherMode cipherMode)
+        {
+            switch (cipherMode)
+            {
+                case CipherMode.CBC:
+                    return Interop.AppleCrypto.PAL_ChainingMode.CBC;
+                case CipherMode.ECB:
+                    return Interop.AppleCrypto.PAL_ChainingMode.ECB;
+                case CipherMode.CTS:
+                    throw new PlatformNotSupportedException(SR.Cryptography_UnsupportedPaddingMode);
+                default:
+                    Debug.Fail($"Unknown cipher mode: {cipherMode}");
+                    throw new CryptographicException(SR.Cryptography_UnknownPaddingMode);
```

Comment:
It wasn't until right now that I noticed that while reading cipher (chaining) mode I used the resources for padding mode.  So.. yes.  I can add resource strings which include the variable.

The other platforms have various roundabout ways of concluding things don't make sense, which is why there isn't a good one already.